modules/renter/repair.go

```diff
@@ -193,7 +194,11 @@ func (r *Renter) managedRepairIteration(rs *repairState) {
 		}
 
 		// Ignore workers that have had an upload failure recently.
-		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown {
+		penalty := uint64(worker.consecutiveUploadFailures)
+		if worker.consecutiveUploadFailures > maxConsecutivePenalty {
+			penalty = uint64(maxConsecutivePenalty)
+		}
+		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown*(1<<penalty) {
```

Comment:
seems slightly brittle to use `<<` since it can only be used for power-of-2 exponentiation. On the other hand, doubling each time seems reasonable enough that it isn't likely to change in the future.

Alternatively, you could move the cooldown time into the worker struct. Then instead of counting failures, you could multiply the cooldown by some factor. That would allow you to exponentiate without requiring a loop.

Regardless, it would be good to have a comment here explaining that the cooldown is exponential. I would also prefer that the "max" value be a duration (e.g. `24*time.Hour`) instead of an exponent; makes it easier to tweak later.